### ISSUE TITLE
refactor repo structure

### ISSUE THREAD

**Issue by mmerickel** [Pylons/pyramid#3022]
1. Move tests out of the pyramid package. They will still be shipped with the sdist but not with the binary wheel.

2. Move the pyramid package into a src subfolder in the repo to conform to best practices in other repos like hupper and pyramid_retry. This is a simple fix to prevent pyramid source from being on the `PYTHONPATH` inadvertently which means we're testing the source instead of the actual sdist built by tox. This also avoids subtle issues in which multiple versions of python are run against the same code.

**Comment by mcdonc** [comment]
+1 on both

**Comment by mmerickel** [comment]
for anyone doing this work, please don't do it until at least https://github.com/Pylons/pyramid/pull/3113 is done

**Comment by mmerickel** [comment]
@mcdonc @tseaver anyone else... are any of you opposed to me running black on the code and adding `black --check -l 79 src/pyramid tests` to `tox -e lint`? if not then I'm going to do it as part of this refactor. I know that at least @bertjwregeer and I are a big +1 on it.

**Comment by mcdonc** [comment]
+0

**Comment by bertjwregeer** [comment]
I'd want to leave the default line length at 88... but I can live with 79, so long as we add a pyproject.toml that includes that as a default setting so I don't have to remember to set it in my editor :-).

**Comment by mcdonc** [comment]
I'm skeeved out by its replacement of ' with ".  But whatever.

**Comment by mmerickel** [comment]
Yeah that’s by far my biggest gripe with it as well. But I’m not sure it’s worth throwing out the whole thing as a result. I’m open to setting the cli flag that preserves quoting but the more flags we set the more ways people can screw it up.

**Comment by mcdonc** [comment]
Given that it's a flag already (it's --skip-string-normalization right?) I might say set it, f it. :)

**Comment by mcdonc** [comment]
When I look at the diff from ``black -l 79 -S pyramid`` I don't see anything that I couldn't live with.

**Comment by bertjwregeer** [comment]
Setting that flag means that no quoting gets modified, which means things like:

```
{'some': "dict", "other": 'whatever'}
```

Is okay...

That bothers me, I'd prefer it just normalized it for me.

Thankfully we can add flags to `pyproject.toml` which will be picked up automatically by a `black` run, so we can avoid people "screwing" it up. Same with the line length.

**Comment by mmerickel** [comment]
the refactoring is done in #3387 and the black config is done in #3388

**Comment by Cykooz** [comment]
How can I use Pyramid tests after that change?

I have a project with extended version of `pyramid.authorization.ACLAuthorizationPolicy`. I am using `TestACLAuthorizationPolicy` test case to testing my implementation:

```python
from pyramid.tests.test_authorization import TestACLAuthorizationPolicy

class TestRestACLAuthorizationPolicy(TestACLAuthorizationPolicy):

    def _getTargetClass(self):
        from ..authorization import RestACLAuthorizationPolicy
        return RestACLAuthorizationPolicy

    # Some addition tests for extended functionality
```

It would be cool if you upload into PyPi another package with all Pyramid tests, e.g. `pyramid.tests`.
